app/controllers/application_controller.rb

```diff
@@ -22,7 +22,14 @@ class ApplicationController < ActionController::Base
   before_action :set_account_specific_connections!
 
   rescue_from Apartment::TenantNotFound do
-    redirect_to splash_path
+    # If you do a return here you'll get a LocalJumpError
```

Comment:
Instead of a return here, could you do:

```ruby
raise ActionController::RoutingError, 'Not Found' unless base_host?
redirect_to splash_path
```

:question: